### PR TITLE
Add Docker support

### PR DESCRIPTION
**Pull request by Menda**
https://github.com/Humanitec/Walhall/issues/20

### CODE REVIEW

File: service_builder/setup_service.py [hunk 1]
```diff
@@ -59,6 +59,18 @@ def _create_app(name_project: str, name_app: str):
         'The app "{}" was successfully created'.format(name_app))
 
 
+def _configure_docker(name_project):
+    filenames = (
+        'Dockerfile', 'Dockerfile.dev', 'Dockerfile.nginx',
```
Comment (jefmoura) [inline service_builder/setup_service.py]: Do we really one keep having this three different Docker files? I agree with @bogdangi that we should have something more close to `production`, I mean one file for all the environments.

File: service_builder/setup_service.py [hunk 2]
```diff
@@ -59,6 +59,18 @@ def _create_app(name_project: str, name_app: str):
         'The app "{}" was successfully created'.format(name_app))
 
 
+def _configure_docker(name_project):
```
Comment (jefmoura) [inline service_builder/setup_service.py]: I didn't get this function. Is it just creating a Dockerfile from a template without changing anything?

File: service_builder/templates/docker/docker-entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+python manage.py migrate
+
+RESULT=$?
+if [ $RESULT -ne 0 ]; then
+    echo $(date -u) " - There was a problem migrating. Setting up the maintenance page"
+    pushd templates/maintenance/; python -m SimpleHTTPServer 8081; popd
+else
+    echo $(date -u) " - Running the server in branch '$branch'"
+
+    gunicorn -b 0.0.0.0:8081 documents-service.wsgi
```
Comment (jefmoura) [inline service_builder/templates/docker/docker-entrypoint.sh]: `documents-service.wsgi`?

**Comment by Menda** [comment]
@jefmoura do not review yet until I tell you

File: service_builder/templates/docker/docker-entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    echo $(date -u) "- Migrating"
+    python manage.py migrate
+
+    RESULT=$?
+    if [ $RESULT -ne 0 ]; then
+        echo $(date -u) "- There was a problem migrating. Exiting."
+        exit 1
+    else
+        echo $(date -u) "- Creating admin user"
+        python manage.py shell -c "from django.contrib.auth.models import User; User.objects.filter(email='admin@sample.com').delete(); User.objects.create_superuser('admin', 'admin@sample.com', 'admin')"
```
Comment (jefmoura) [inline service_builder/templates/docker/docker-entrypoint.sh]: I think it's better to generate a random password and print on the screen. Don't you think?
https://www.howtogeek.com/howto/30184/10-ways-to-generate-a-random-password-from-the-command-line/

Comment (jefmoura) [inline service_builder/templates/docker/docker-entrypoint.sh]: `sample.com` exists, so use `example.com` instead.

File: service_builder/templates/docker/docker-entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+        echo $(date -u) "- Running the development server"
+        PYTHONUNBUFFERED=1 python manage.py runserver 0.0.0.0:8080
+    fi
+else
+    echo $(date -u) "- Running the server"
+    gunicorn -b 0.0.0.0:8080 {{ name_project }}.wsgi
```
[the remark this answers is not in the document]
Comment (Menda) [reply service_builder/templates/docker/docker-entrypoint.sh]: pdb support and server autoreload when there are changes mainly.

File: service_builder/templates/docker/docker-entrypoint.sh (hunk 4 above)
Comment (Menda) [reply service_builder/templates/docker/docker-entrypoint.sh]: This is only for localhost, I don't think it's big deal to have admin/admin

Comment (Menda) [reply service_builder/templates/docker/docker-entrypoint.sh]: OK!

**Comment by Menda** [comment]
@jefmoura you can check again

File: service_builder/templates/docker/docker-entrypoint.sh (hunk 4 above)
Comment (jefmoura) [reply service_builder/templates/docker/docker-entrypoint.sh]: - It's fine. We really don't need to create a complex password for local env.
- You don't show on the screen that `admin` user with `admin` password was created. There's nothing in the documentation as well. How can I know that?

Comment (Menda) [reply service_builder/templates/docker/docker-entrypoint.sh]: There is info in the documentation. See: https://github.com/Humanitec/django-service-wizard/pull/6/files#diff-92c75cf4e92017a0a409008b6ca0c6b5R22

Comment (jefmoura) [reply service_builder/templates/docker/docker-entrypoint.sh]: Ok. I didn't see that :sweat_smile:

**Pull request merged**